### PR TITLE
Refactor the get DS leader peer into a function

### PR DESCRIPTION
**Pull request by ShengguangXiao**
## Description

Refactor the get DS leader peer into a function

Add Node::GetDSLeaderPeer function, so it can be reused in multiple places. Easy to maintain.

https://github.com/Zilliqa/Issues/issues/358

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

**Comment by KaustubhShamshery** [comment]
Can you also change in FinishRejoinAsDS

File: src/libLookup/Lookup.cpp [hunk 1]
```diff
@@ -3168,7 +3167,7 @@ void Lookup::SendTxnPacketToNodes(uint32_t numShards) {
       {
         lock_guard<mutex> g(m_mediator.m_ds->m_mutexShards);
         auto it = m_mediator.m_ds->m_shards.at(i).begin();
-        //[FIX-ME]: Lookup sends to NUM_NODES_TO_SEND_LOOKUP + Leader
+        // Lookup sends to NUM_NODES_TO_SEND_LOOKUP + Leader
```
Comment (sandipbhoir) [inline src/libLookup/Lookup.cpp]: FIX ME - was added here to fix the number of total nodes to send including leader. right now its NUM_NODES_TO_SEND_LOOKUP + Leader which needs t be fixed eventually.

File: src/libNode/PoWProcessing.cpp [hunk 2]
```diff
@@ -287,33 +287,13 @@ bool Node::SendPoWResultToDSComm(const uint64_t& block_num,
       break;
     }
   }
-  //[FIX-ME] Send to PoW PACKET_SENDERS + 1
+  // Send to PoW PACKET_SENDERS + DS Leader
```
Comment (sandipbhoir) [inline src/libNode/PoWProcessing.cpp]: FIX ME - was added here to fix the number of total nodes to send including leader. right now its NUM_NODES_TO_SEND_LOOKUP + Leader which needs t be fixed eventually.

File: src/libNode/Node.cpp [hunk 3]
```diff
@@ -1668,3 +1669,42 @@ std::string Node::GetActionString(Action action) const {
...
+    if (!GUARD_MODE) {
+      leader_id = lastBlockHash % dsCommittee.size();
+    } else {
+      leader_id = lastBlockHash % Guard::GetInstance().GetNumOfDSGuard();
+    }
+    dsLeaderPeer = dsCommittee.at(leader_id).second;
+    LOG_EPOCH(INFO, to_string(epochNumber).c_str(),
+              "lastBlockHash " << lastBlockHash << ", current ds leader id "
+                               << leader_id << ", Peer "
+                               << std::string(dsLeaderPeer));
```
Comment (sandipbhoir) [inline src/libNode/Node.cpp]: Do we need tostring() for printing peer?

Comment (ShengguangXiao) [reply src/libNode/Node.cpp]: No need, i fixed it already.

**Pull request merged**